chain/ethereum/src/runtime/runtime_adapter.rs

```diff
@@ -0,0 +1,206 @@
+use std::{sync::Arc, time::Instant};
+
+use crate::{
+    network::EthereumNetworkAdapters, Chain, DataSource, EthereumAdapter, EthereumAdapterTrait,
+    EthereumContractCall, EthereumContractCallError,
+};
+use anyhow::{Context, Error};
+use blockchain::HostFn;
+use ethabi::{Address, Token};
+use graph::{
+    blockchain::{self, BlockPtr, DataSource as _, HostFnCtx},
+    cheap_clone::CheapClone,
+    components::ethereum::NodeCapabilities,
+    prelude::{EthereumCallCache, Future01CompatExt, MappingABI},
+    runtime::{asc_get, asc_new, AscPtr, HostExportError},
+    semver::Version,
+    slog::{info, trace, Logger},
+};
+use graph_runtime_wasm::asc_abi::class::{AscEnumArray, EthereumValueKind};
+
+use super::abi::{AscUnresolvedContractCall, AscUnresolvedContractCall_0_0_4};
+
+pub struct RuntimeAdapter {
+    pub(crate) eth_adapters: Arc<EthereumNetworkAdapters>,
+    pub(crate) call_cache: Arc<dyn EthereumCallCache>,
+}
+
+impl blockchain::RuntimeAdapter<Chain> for RuntimeAdapter {
+    fn host_fns(&self, ds: &DataSource) -> Result<Vec<HostFn>, Error> {
```

Comment:
Would it make sense for the `HostFn` vector to be a `HashMap` using the `function` names as keys?

Reply:
I guess that would guarantee that the function names are unique, but besides that it wouldn't gain us much. I like the simplicity of good ol' `Vec`.

And `HashMap` iteration order is not deterministic, so I'd use `BTreeMap`.